### PR TITLE
use pod networking during docker build

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/13208

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton broken until we can reconcile the level of the docker dependency that the new fsouza go-client needs.

**Comment by bparees** [comment]
[testextended][extended:core(builds)]

**Comment by bparees** [comment]
@smarterclayton ptal, guessing there's some special naming convention you'd like me to use for the vendoring commit.

I also had to manually delete the runc godep was trying to add...despite being the exact same commit as what origin was already vendoring (so i'm not even sure why it wanted to add it), it was causing compilation problems due to duplicate definitions... so something isn't properly resolving the gopath it seems.

There are a bunch of other deps in here that seem to be common between origin/vendor and origin/pkg/build/vendor, so again not sure what godeps is doing (but unlike runc, they don't cause compilation problems).  I can manually delete them all (like i did for runc) but before i do that i figured i'd run it by you.

**Comment by bparees** [comment]
the process i followed was roughly:

```
cd origin
godep restore
git checkout newer fsouza and docker
cd pkg/builder
godep save github.com/fsouza/go-client/...
# manually remove opencontainers/runc vendored dep
```

**Comment by bparees** [comment]
```
continuous-integration/openshift-jenkins/testextended SUCCESS (https://ci.openshift.redhat.com/jenkins/job/test_pr_origin_extended/1166/) (Base Commit: b4652ac) (Extended Tests: core(builds))
```

[testextended][extended:core(image_ecosystem)]

File: pkg/build/Godeps/Godeps.json [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+		{
+			"ImportPath": "github.com/fsouza/go-dockerclient/testing",
+			"Comment": "0.2.1-1283-ga45f2c4",
+			"Rev": "a45f2c476754603041aeb54751207d5e5ec30b77"
+		},
+		{
+			"ImportPath": "github.com/gorilla/context",
+			"Rev": "215affda49addc4c8ef7e2534915df2c8c35c6cd"
+		},
+		{
```
Comment (smarterclayton) [inline pkg/build/Godeps/Godeps.json]: I don't know that you need this package

File: pkg/build/Godeps/Godeps.json [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+			"ImportPath": "github.com/gorilla/mux",
+			"Rev": "8096f47503459bcc74d1f4c487b7e6e42e5746b5"
+		},
+		{
+			"ImportPath": "github.com/hashicorp/go-cleanhttp",
+			"Rev": "3573b8b52aa7b37b9358d966a898feb387f62437"
+		},
+		{
```
Comment (smarterclayton) [inline pkg/build/Godeps/Godeps.json]: Shouldn't need this anymore, context is in go proper

File: pkg/build/Godeps/Godeps.json [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+			"ImportPath": "golang.org/x/net/context",
+			"Rev": "e90d6d0afc4c315a0d87a568ae68577cc15149a0"
+		},
+		{
+			"ImportPath": "golang.org/x/net/context/ctxhttp",
+			"Rev": "e90d6d0afc4c315a0d87a568ae68577cc15149a0"
+		},
+		{
```
Comment (smarterclayton) [inline pkg/build/Godeps/Godeps.json]: Try removing this and below - I don't think you need it.

**Comment by smarterclayton** [comment]
Vendor commits should be `bump(PACKAGE):COMMIT` (package is the primary package you're trying to get , GitHub.com/fsouza/go-dockerclient)

**Comment by smarterclayton** [comment]
So many new lines of code...

How much bigger does the openshift binary get before and after?

**Comment by bparees** [comment]
> Vendor commits should be bump(PACKAGE):COMMIT (package is the primary package you're trying to get , GitHub.com/fsouza/go-dockerclient)

ok, i wasn't sure how to handle the fact that it was bumping/adding other vendored packages.

Anyway it looks like you're bumping the main fsouza package after all, so should I just dump all this vendoring?

File: pkg/build/Godeps/Godeps.json (hunk 2 above)
Comment (bparees) [inline pkg/build/Godeps/Godeps.json]: fsouza imports it explicitly though: 

```
pkg/build/vendor/github.com/fsouza/go-dockerclient/client.go:36:2: cannot find package "github.com/hashicorp/go-cleanhttp" 
```

**Comment by smarterclayton** [comment]
No you still have to - I'm just going to ensure I have to do nothing except bring image builder into your vendor.

File: pkg/build/Godeps/Godeps.json (hunk 1 above)
Comment (smarterclayton) [inline pkg/build/Godeps/Godeps.json]: If you dump fsouza/go-dockerclient/testing you can remove these.  Do you need testing?

Comment (bparees) [inline pkg/build/Godeps/Godeps.json]: i don't think so, just forgot to specify -t on godeps.

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
openshift binary sizes with your suggested removals:
new: 267megs
master: 266megs

File: pkg/build/Godeps/Godeps.json [hunk 4]
```diff
@@ -0,0 +1,176 @@
+{
+	"ImportPath": "github.com/openshift/origin/pkg/build",
+	"GoVersion": "go1.7",
+	"GodepVersion": "v74",
+	"Packages": [
+		"github.com/fsouza/go-dockerclient/..."
+	],
+	"Deps": [
+		{
+			"ImportPath": "github.com/Azure/go-ansiterm",
```
Comment (smarterclayton) [inline pkg/build/Godeps/Godeps.json]: I think you can purge these first three too.

Comment (smarterclayton) [inline pkg/build/Godeps/Godeps.json]: Unless we depend on them in s2i or something.

**Comment by bparees** [comment]
assuming this passes tests i'll squash+fix the commit text.

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/9285
[test]

**Comment by smarterclayton** [comment]
Lgtm, squash and rename the bump commit and I'll merge.

**Comment by bparees** [comment]
i'd like to know what's going on w/ all the DB replica tests suddenly failing but i guess it wasn't introduced by this.

**Comment by bparees** [comment]
@smarterclayton commits fixed.  ready for merge.

**Comment by bparees** [comment]
```
continuous-integration/openshift-jenkins/testextended SUCCESS (https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended/21/) (Base Commit: 209f100) (Extended Tests: core(builds))
```

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**